server/app/filters/NonUserRoutes.java

```diff
@@ -0,0 +1,29 @@
+package filters;
+
+import com.google.common.collect.ImmutableList;
+import play.mvc.Http;
+
+/**
+ * Path prefixes and static paths for HTTP routes that may not involve a user session and can be
+ * safely skipped by filters that act on the user session or account. The prefixes all should not
+ * involve a session, and the programs path may or may not, but is included here to allow it not to.
+ */
+final class NonUserRoutes {
```

Comment:
Since we're including `/programs` which is a user route, I think it might be confusing to call this `NonUserRoutes`. I think we should either include `/programs` in the `CiviFormProfileFilter` explicitly or change the name of this. I think other filters also use `NonUserRoutes` (maybe the settings filter?) so I would opt to include `/programs` explicity in `CiviFormProfileFilter` and remove it from this class.

Reply:
I was thinking to include `/` in non user routes since it's not a route a user would see. It just directs them based on their profile if they have one. But I don't feel strongly about it, and the same concern could apply about unintended consequences for other filters using the non user routes filter.